src/ui/public/filter_bar/filter_bar_click_handler.js

```diff
@@ -57,11 +60,20 @@ export function FilterBarClickHandlerProvider(Notifier) {
         }
 
         filters = dedupFilters($state.filters, uniqFilters(filters), { negate: true });
-        // We need to add a bunch of filter deduping here.
+
         if (!simulate) {
-          $state.$newFilters = filters;
+          if ($state.query.language === 'lucene') {
+            $state.$newFilters = filters;
+          }
+          else if ($state.query.language === 'kuery') {
+            addFiltersToKuery($state, filters)
+            .then(() => {
+              if (_.isFunction($state.save)) {
+                $state.save();
+              }
+            });
```

Comment:
filter_bar_click_handler now handles updating kuery queries itself, as it probably should have from the beginning.